src/data/nat/basic.lean

```diff
@@ -745,6 +745,12 @@ lemma lt_pow_self {p : ℕ} (h : 1 < p) : ∀ n : ℕ, n < p ^ n
   n + 1 < p^n + 1 : nat.add_lt_add_right (lt_pow_self _) _
     ... ≤ p ^ (n+1) : pow_lt_pow_succ h _
 
+lemma pow_monotonic { x m n : ℕ } ( ha : 2 ≤ x ) ( k : x^m ≤ x^n ) : m ≤ n :=
+le_of_not_lt (λ r, not_le_of_lt (pow_lt_pow_of_lt_right ha r) k)
+
+lemma pow_right_inj { x m n : ℕ } ( ha : 2 ≤ x ) ( k : x^m = x^n ) : m = n :=
```

Comment:
These are often more usefully stated as an iff. Proving `x^m = x^n ↔ m = n` allows the lemma to be used by `rw` and `simp`

Reply:
Thanks for the suggestion! I'll adjust this to type
```lean
{x m n : ℕ} (ha : 2 ≤ x): x^m = x^n ↔ m = n := ...
```?
Is it better to prove the iff version, and deduce this original using `.2`, or to leave this proof as-is, and deduce an iff afterwards?